src/modules/activities/profile/partials/ProfileForm.tsx

```diff
@@ -0,0 +1,225 @@
+import { Button, Modal, Spinner } from 'flowbite-react';
+import { useForm } from 'react-hook-form';
+import { joiResolver } from '@hookform/resolvers/joi';
+import { useState, useEffect } from 'react';
+import {
+  HiPencil,
+  HiChevronRight,
+  HiChevronLeft,
+  HiX,
+  HiCheck,
+} from 'react-icons/hi';
+import { toast } from 'react-toastify';
+
+import PersonalInformation from './information/PersonalInformation';
+import AddressInformation from './information/AddressInformation';
+import BankInformation from './information/BankInformation';
+
+import profileSchema from '@/api/schema/profile';
+import {
+  useAppDispatch,
+  useAppSelector,
+} from '@/modules/_partials/hooks/useRedux';
+import { updateProfile } from '@/redux/features/profile/profileSlice';
+
+const ProfileForm = () => {
+  const dispatch = useAppDispatch();
+  const { user, updateError, isUpdating } = useAppSelector(
+    state => state.profile,
+  );
+
+  const profile = user || {};
+
+  const [schema, setSchema] = useState(profileSchema.personal);
+  const {
+    register,
+    reset,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({
+    resolver: joiResolver(schema),
+  });
+  const [profileData, setProfileData] = useState({
+    displayName: profile.displayName,
+    firstName: profile.firstName,
+    lastName: profile.lastName,
+    gender: profile.gender,
+    address: profile.address || {},
+    bank: profile.bank || {},
+  });
+  const [isOpen, setIsOpen] = useState(false);
+  const [step, setStep] = useState(1);
+  const onClose = () => {
+    if (isUpdating) return;
+    setIsOpen(false);
+  };
+
+  const togglePopup = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const renderSteps = () => {
+    switch (step) {
+      case 1:
+        return (
+          <PersonalInformation register={register} errors={errors} />
```

Comment:
instead of doing this
```
<Component  prop1={props1} prop2={prop2} />
```
you can destructure the prop that matches the variable name
```
<Component  {...{ props1, prop2 }} />
```
it is not a flow or bad in any way, it just improves the code's readability

Reply:
This is good idea.